packages/jest-runner/__typetests__/jest-runner.test.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {UnsubscribeFn} from 'emittery';
+import {expectType} from 'tsd-lite';
+import type {Test, TestEvents} from '@jest/test-result';
+import type {Config} from '@jest/types';
+import {CallbackTestRunner, EmittingTestRunner} from 'jest-runner';
+import type {
+  OnTestFailure,
+  OnTestStart,
+  OnTestSuccess,
+  TestRunnerContext,
+  TestRunnerOptions,
+  TestWatcher,
+} from 'jest-runner';
+
+const globalConfig = {} as Config.GlobalConfig;
+const runnerContext = {} as TestRunnerContext;
+
+class CallbackRunner extends CallbackTestRunner {
+  async runTests(
```

Comment:
I was hoping that `override` here in a test could catch some drift (hence #12648). Apparently `override` has no effect in this case, because methods are marked `abstract`.

In the other hand, `abstract` requires to implement those methods in a derived class. So to have this test passing is still good.